### ISSUE TITLE
Accessing a resource granted by a resource access request should not be predicated upon losing access to all other resources

### ISSUE THREAD

**Issue by ravicious** [gravitational/teleport#36800]
**What would you like Teleport to do?**

The user should be able to access resources that a resource access request grants them access to _and_ resources that are available to the user by default (without assuming any access request).

**What problem does this solve?**

Resource access requests are more friendly to end users than role access requests. The user simply needs to know which resource they want to access, without having to know which role grants access to the resource.

However, they can be difficult to work with because assuming a resource access request means that [the user can access only resources listed in the access request](https://goteleport.com/docs/access-controls/access-requests/resource-requests/#step-88-resume-regular-access).

Imagine a scenario where there are two resources, "foo" that is available to the user without any access requests and "bar" that is available through a resource request, and the user wants to open two shell sessions and access both resources. If they first assume a request for bar, they won't be able to access foo unless they drop the access request.

This is hard to work with in a CLI setting where the user might have multiple shell sessions open and would like to connect to both resources.

**If a workaround exists, please include it.**

One workaround is to request a specific role manually. This is predicated on knowing which role grants access to the resource I want to access.

Another might be to log in, assume an access request, connect to the resource from the access request and then in another shell session drop the access request and connect to another resource. Or vice versa, first connect to a resource not gated by an access request and then assume the request and connect to the resource gated by an access request. This is still hard to work with in scenarios where the user needs to repeatedly establish new connections to either of the resources.

---

Related:

* [A discussion in a customer support channel](https://goteleport.slack.com/archives/C0582MBSMHN/p1694612469633529) where the customer would prefer if there was a choice between a resource request and a role request when accessing it with CLI. The discussion touches a bit on how resource access requests play into the rest of the Teleport architecture and what the caveats are.
* [A support ticket](https://gravitational.zendesk.com/agent/tickets/9637) where the default behavior of resource access requests is surprising for the customer.

**Comment by zmb3** [comment]
Duplicate of #28304
